planner/core/logical_plan_builder.go

```diff
@@ -2657,6 +2657,11 @@ func (b *PlanBuilder) buildSelect(ctx context.Context, sel *ast.SelectStmt) (p L
 		gbyCols                       []expression.Expression
 	)
 
+	// set for update read to true before building result set node
+	if sel.LockTp != ast.SelectLockNone {
```

Comment:
We may need to abstract a function for it, actually we do not `for update` read using `lock in share mode` as the [doc ](https://docs.pingcap.com/zh/tidb/stable/pessimistic-transaction#%E5%92%8C-mysql-innodb-%E7%9A%84%E5%B7%AE%E5%BC%82)described.

Reply:
May this function in parser?

Reply:
For this pr we could put it in TiDB repo.